src/libasr/codegen/llvm_utils.cpp

```diff
@@ -373,6 +373,129 @@ namespace LFortran {
         shift_end_point_by_one(list);
     }
 
+    llvm::Value* LLVMList::find_item_position(llvm::Value* list,
+                                    llvm::Value* item, llvm::Type* el_type) {
+        llvm::Value* current_end_point = LLVM::CreateLoad(*builder,
+                                        get_pointer_to_current_end_point(list));
+        llvm::AllocaInst *i = builder->CreateAlloca(el_type, nullptr);
+        LLVM::CreateStore(*builder, llvm::ConstantInt::get(
+                                    context, llvm::APInt(32, 0)), i);
+        llvm::Value* tmp = nullptr;
+
+        /* Equivalent in C++:
+         * int i = 0;
+         * while(list[i] != item && end_point > i) {
+         *     i++;
+         * }
+         *
+         * if (i == end_point) {
+         *    std::cout << "Element: `item` not found in the list";
+         * }
+         */
+
+        llvm::BasicBlock *loophead = llvm::BasicBlock::Create(context, "loop.head");
+        llvm::BasicBlock *loopbody = llvm::BasicBlock::Create(context, "loop.body");
+        llvm::BasicBlock *loopend = llvm::BasicBlock::Create(context, "loop.end");
+
+        // head
+        llvm_utils->start_new_block(loophead);
+        {
+            llvm::Value *cond = builder->CreateAnd(builder->CreateICmpNE(
+                read_item(list, LLVM::CreateLoad(*builder, i), false), item),
+                builder->CreateICmpSGT(current_end_point,
+                                       LLVM::CreateLoad(*builder, i)));
+            builder->CreateCondBr(cond, loopbody, loopend);
+        }
+
+        // body
+        llvm_utils->start_new_block(loopbody);
+        {
+            tmp = builder->CreateAdd(
+                        LLVM::CreateLoad(*builder, i),
+                        llvm::ConstantInt::get(context, llvm::APInt(32, 1)));
+            LLVM::CreateStore(*builder, tmp, i);
+        }
+        builder->CreateBr(loophead);
+
+        // end
+        llvm_utils->start_new_block(loopend);
+
+        llvm::Value* cond = builder->CreateICmpEQ(
+                              LLVM::CreateLoad(*builder, i), current_end_point);
+        llvm::Function *fn = builder->GetInsertBlock()->getParent();
+        llvm::BasicBlock *thenBB = llvm::BasicBlock::Create(context, "then", fn);
+        llvm::BasicBlock *elseBB = llvm::BasicBlock::Create(context, "else");
+        llvm::BasicBlock *mergeBB = llvm::BasicBlock::Create(context, "ifcont");
+        builder->CreateCondBr(cond, thenBB, elseBB);
+
+        builder->SetInsertPoint(thenBB);
+            // "Element: `item` not found in the list"
+            // replace this with Diagnostic Error message.
+            // Print the `item` in the error message.
```

Comment:
I need to replace this with an Exception saying "Element not found in the list", right?
For that I added `throw LCompilersException()`, I don't know what the problem is, but even when the `cond` is false the stacktrace appears.

Reply:
You mean, The non-existing element has to be handled before it comes to the Back-end? (Like in the Python_ast_to_asr or something similar)

Reply:
How can you handle any input element at compile time when its value will be known only at run-time? I can call, `list.remove(x)` where is `x` is available only at runtime (say loaded from a file, comes from C runtime or whatever).

Reply:
Sorry my bad, I got it!
How do I handle the above issue?

Reply:
Well I don’t know for sure. So, LPython raises AssertionError if assert check fails. We can do something similar for ValueError here. It won’t be be as good as CPython but anyways it would be something working and familiar to the users. @certik thoughts?

Reply:
> In here if the item cannot be found we simply do whatever the semantics of the operation is. That is, we have to choose what happens if ListRemove can't find the item. I would do what we do for if ArrayItem is out of bounds --- we should check it (optionally) and stop the program (with a nice stacktrace eventually).

@certik commented this in the previous discussion.

Reply:
For now just stop the program at runtime with an error. Later we'll add some more unified and better runtime error handling.